runtime/vm/ContinuationHelpers.cpp

```diff
@@ -88,19 +96,41 @@ createContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 	return result;
 }
 
+void
+synchronizeWithConcurrentGCScan(J9VMThread *currentThread, J9VMContinuation *continuation)
+{
+	volatile uintptr_t *localAddr = &continuation->state;
+	/* atomically 'or' (not 'set') continuation->state  with currentThread */
+	uintptr_t oldContinuationState = VM_AtomicSupport::bitOr(localAddr, (uintptr_t)currentThread);
+
+	if (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(oldContinuationState)) {
+		/* currentThread was low tagged (GC was already in progress) */
+		internalReleaseVMAccess(currentThread);
+		randomSleep();
+
+		omrthread_monitor_enter(currentThread->publicFlagsMutex);
+		while (VM_VMHelpers::isConcurrentlyScannedFromContinuationState(*localAddr)) {
+			/* GC is still concurrently scanning the continuation(currentThread was still low tagged) */
+			omrthread_monitor_wait(currentThread->publicFlagsMutex);
+		}
+		omrthread_monitor_exit(currentThread->publicFlagsMutex);
+
+		randomSleep();
+		internalAcquireVMAccess(currentThread);
+	}
+}
+
 BOOLEAN
 enterContinuation(J9VMThread *currentThread, j9object_t continuationObject)
 {
 	BOOLEAN result = TRUE;
 	jboolean started = J9VMJDKINTERNALVMCONTINUATION_STARTED(currentThread, continuationObject);
 	J9VMContinuation *continuation = J9VMJDKINTERNALVMCONTINUATION_VMREF(currentThread, continuationObject);
 
-	Assert_VM_Null(currentThread->currentContinuation);
-
+	synchronizeWithConcurrentGCScan(currentThread, continuation);
```

Comment:
you can add a comment: 
let GC know we are mounting, so they don't need to scan us, or if there is already ongoing scan wait till it's complete